examples/views/textbox.rs

```diff
@@ -1,22 +1,13 @@
 use vizia::prelude::*;
 
-#[derive(Lens)]
+#[derive(Lens, Ray)]
 pub struct AppData {
     text: String,
 }
 
-#[derive(Debug)]
-pub enum AppEvent {
-    SetText(String),
-}
-
 impl Model for AppData {
     fn event(&mut self, _: &mut EventContext, event: &mut Event) {
-        event.map(|app_event, _| match app_event {
-            AppEvent::SetText(text) => {
-                self.text = text.clone();
-            }
-        });
+        event.take::<AppDataRay>().map(|ray| ray.apply(self));
```

Comment:
Should we use this here or make a dedicated example for it?

Reply:
So it depends on the way we want to structure our teaching. I think we should make it so that we introduce rays as a basic building block, after explaining how to do the same thing with "standard" events/messages.

Reply:
Which is to say, I think we should convert examples to using rays as applicable and as we come across them

Reply:
Okay, seems reasonable to me.